### PR TITLE
feat(consul-provider): If service ip is nil then use node ip

### PR DESCRIPTION
**Pull request by samber**
#294 

Summary: In consul, Service.Address is optional. If equal to nil, then use the Node.Address

### CODE REVIEW

**Comment by emilevauge** [comment]
@samber, Could you include a bug fix for #309 in this PR? Thanks ;)

**Comment by keis** [comment]
thank you for this @samber 

A small nit but I think it would look better to have a getAddress helper function to keep the template tidy

**Comment by samber** [comment]
I don't have time to write tests before weekend ;-)
See you next week

File: provider/consul_catalog.go [hunk 1]
```diff
@@ -123,6 +123,27 @@ func (provider *ConsulCatalog) getFrontendRule(service serviceUpdate) string {
...
+ gliderslab/registrator registers services with an "udp" tag if not tcp
+*/
+func (provider *ConsulCatalog) getBackendName(node *api.ServiceEntry) string {
+	service_name := strings.Replace(node.Service.Service, ".", "-", -1) + "--" + strings.Replace(node.Service.Address, ".", "-", -1) + "--" + string(node.Service.Port)
+	for _, tag := range node.Service.Tags {
+		if tag == "udp" {
+			return service_name + "--udp"
+		}
+	}
+	return service_name
```
Comment (emilevauge) [inline provider/consul_catalog.go]: I would rather **always** suffix backend names with tags.
```
return service_name +  "-"  + strings.join(node.Service.Tags, "-")
```
WDYT?

Comment (samber) [inline provider/consul_catalog.go]: Done ;-)

**Comment by emilevauge** [comment]
LGTM!
Needs a rebase :)

**Comment by samber** [comment]
Done ;-)

**Comment by vdemeester** [comment]
@samber another one (sorry :sweat:)

**Comment by keis** [comment]
hey guys :sweat_smile:  anything I could help out with to get this merged?

**Comment by emilevauge** [comment]
Yes, with a rebase ^^

**Comment by samber** [comment]
May this new rebase help you ? ;-)

youhou

**Comment by emilevauge** [comment]
Thanks @samber  :)
/cc @vdemeester

**Comment by vdemeester** [comment]
LGTM :angel:

**Pull request merged**